textattack/transformations/word_swap_masked_lm.py

```diff
@@ -0,0 +1,209 @@
+                    replacement_words.append(token)
+            return replacement_words
+        else:
+            # Word to replace is tokenized as multiple sub-words
+            top_preds = [top_pred_ids[i] for i in target_ids_pos]
+            products = itertools.product(*top_preds)
+            combination_results = []
+            # Original BERT-Attack implement uses cross-entropy loss
+            cross_entropy_loss = torch.nn.CrossEntropyLoss(reduction="none")
+            target_ids_pos_tensor = torch.Tensor(target_ids_pos).long()
+            word_tensor = torch.zeros(len(target_ids_pos), dtype=torch.long)
+            for bpe_tokens in products:
+                for i in range(len(bpe_tokens)):
+                    word_tensor[i] = bpe_tokens[i]
+
+                logits = torch.index_select(masked_lm_logits, 0, target_ids_pos_tensor)
+                loss = cross_entropy_loss(logits, word_tensor)
+                perplexity = torch.exp(torch.mean(loss, dim=0)).item()
+                word = "".join(
+                    self._lm_tokenizer.convert_ids_to_tokens(word_tensor)
+                ).replace("##", "")
+                if check_if_word(word):
+                    combination_results.append((word, perplexity))
+            # Sort to get top-K results
+            sorted(combination_results, key=lambda x: x[1])
+            top_replacements = [
+                x[0] for x in combination_results[: self.max_candidates]
+            ]
+            return top_replacements
+
+    def _get_replacement_words(self, current_text, index, extra_args=None):
+        if self.method == "bae":
+            return self._bae_replacement_words(current_text, index)
+        elif self.method == "bert-attack":
+            return self._bert_attack_replacement_words(current_text, index, extra_args)
+        else:
+            raise ValueError(f"Unrecognized value {self.method} for `self.method`.")
+
+    def _get_transformations(self, current_text, indices_to_modify):
+        extra_args = {}
+        if self.method == "bert-attack":
+            tokenized_text = self._lm_tokenizer.encode_plus(
+                current_text.text,
+                max_length=self.max_length,
+                pad_to_max_length=True,
+                return_offsets_mapping=True,
+            )
+
+            current_ids = torch.tensor([tokenized_text["input_ids"]]).to(utils.device)
+            token2char_offset = tokenized_text["offset_mapping"]
+            with torch.no_grad():
+                pred_probs = self._language_model(
+                    current_ids, token_type_ids=self._segment_tensor
+                )[0][0]
+            top_probs, top_ids = torch.topk(pred_probs, self.max_candidates)
+            extra_args["top_pred_ids"] = top_ids.cpu()
+            extra_args["masked_lm_logits"] = pred_probs.cpu()
+            extra_args["current_ids"] = tokenized_text["input_ids"]
+            extra_args["token2char_offset"] = token2char_offset
```

Comment:
can you pass these as optional parameters? the dict is too error-prone

Reply:
Yes I think `**kwargs` would work better